web-app/django/VIM/apps/instruments/views/update_umil_db.py

```diff
@@ -0,0 +1,132 @@
+"""Django view to handle user input to UMIl database"""
+
+import json
+from django.contrib.auth.decorators import login_required
+from django.views.decorators.http import require_POST, require_http_methods
+from django.http import JsonResponse
+from django.shortcuts import redirect
+from django.urls import reverse
+from VIM.apps.instruments.models import Instrument, Language, InstrumentName
+
+
+@login_required
+@require_POST
+def add_name(request):
+    """
+    View to add new instrument names to UMIL database.
+
+    This view expects a POST request with the JSON body:
+    {
+        "wikidata_id": "Q12345",
+        "entries": [
+            {
+                "language": "en",
+                "name": "English label",
+                "source": "Source name",
+            },
+            {
+                "language": "fr",
+                "name": "French label",
+                ...
+            }
+        ],
+    }
+
+    Returns:
+        JsonResponse: JSON response with status and message
+    """
+    if request.method == "POST":
+        # Parse the JSON request body
+        data = json.loads(request.body)
+        wikidata_id = data.get("wikidata_id")
+        entries = data.get("entries", [])
+        if not wikidata_id or not entries:
+            return JsonResponse(
+                {
+                    "status": "error",
+                    "message": "Missing required data",
+                }
+            )
+    try:
+        # Fetch the instrument from the database
+        instrument = Instrument.objects.get(wikidata_id=wikidata_id)
```

Comment:
This try / except is too big. Catch the DoesNotExist exception immediately after calling this .get(), and return your error.